File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/BillingCycle.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public enum BillingCycle {
    Monthly,
    Yearly;

    public String toWire() {
        return switch (this) {
            case Monthly -> "monthly";
            case Yearly -> "yearly";
        };
    }

    public static BillingCycle fromWire(String value) {
        if (value == null || value.isBlank()) throw new IllegalArgumentException("BillingCycle is required.");
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "monthly" -> Monthly;
            case "yearly" -> Yearly;
            default -> throw new IllegalArgumentException("Unknown BillingCycle: " + value);
        };
    }
}

File: apps/backend/src/Common/Domain/ValueObjects/BillingCycle.cs
namespace OffceOs.Domain.Common.ValueObjects;

public enum BillingCycle
{
    Monthly,
    Yearly,
}

public static class BillingCycleExtensions
{
    public static string ToStorageString(this BillingCycle cycle) => cycle switch
    {
        BillingCycle.Monthly => "monthly",
        BillingCycle.Yearly => "yearly",
        _ => throw new ArgumentOutOfRangeException(nameof(cycle)),
    };

    public static BillingCycle ToBillingCycle(this string value) => value switch
    {
        "monthly" => BillingCycle.Monthly,
        "yearly" => BillingCycle.Yearly,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown billing cycle: {value}"),
    };
}
